tests/lib/rules/no-global-regexp-flag-in-query.test.ts

```diff
@@ -0,0 +1,189 @@
+    `,
+    `
+        const text = 'hello';
+        /hello/g.test(text)
+        text.match(/hello/g)
+    `,
+    `
+        const text = somethingElse()
+        /hello/g.test(text)
+        text.match(/hello/g)
+    `,
+    `
+        import somethingElse from 'somethingElse'
+        somethingElse.lookup(/hello/g)
+    `,
+    `
+        import { screen } from '@testing-library/dom'
+        screen.notAQuery(/hello/g)
+    `,
+    `
+        import { screen } from '@testing-library/dom'
+        screen.notAQuery('button', {name: /hello/g})
+    `,
+    `
+        const utils = render(<Component/>)
+        utils.notAQuery('button', {name: /hello/i})
+    `,
+    `
+        const utils = render(<Component/>)
+        utils.notAQuery(/hello/i)
+    `,
+  ],
+  invalid: [
+    {
+      code: `
+        import { screen } from '@testing-library/dom'
+        screen.getByText(/hello/g)`,
+      errors: [
+        {
+          messageId: 'noGlobalRegExpFlagInQuery',
+          line: 3,
+          column: 26,
+        },
+      ],
+      output: `
+        import { screen } from '@testing-library/dom'
+        screen.getByText(/hello/)`,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/dom'
+        screen.findByRole('button', {name: /hellogg/g})`,
+      errors: [
+        {
+          messageId: 'noGlobalRegExpFlagInQuery',
+          line: 3,
+          column: 44,
+        },
+      ],
+      output: `
+        import { screen } from '@testing-library/dom'
+        screen.findByRole('button', {name: /hellogg/})`,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/dom'
+        screen.findByRole('button', {otherProp: true, name: /hello/g})`,
+      errors: [
+        {
+          messageId: 'noGlobalRegExpFlagInQuery',
+          line: 3,
+          column: 61,
+        },
+      ],
+      output: `
+        import { screen } from '@testing-library/dom'
+        screen.findByRole('button', {otherProp: true, name: /hello/})`,
+    },
+    {
+      code: `
+        const utils = render(<Component/>)
+        utils.findByRole('button', {name: /hello/ig})`,
+      errors: [
+        {
+          messageId: 'noGlobalRegExpFlagInQuery',
+          line: 3,
+          column: 43,
+        },
+      ],
+      output: `
+        const utils = render(<Component/>)
+        utils.findByRole('button', {name: /hello/i})`,
+    },
+    {
+      code: `
+        const {queryAllByLabelText} = render(<Component/>)
+        queryAllByLabelText(/hello/gi)`,
+      errors: [
+        {
+          messageId: 'noGlobalRegExpFlagInQuery',
+          line: 3,
+          column: 29,
+        },
+      ],
+      output: `
+        const {queryAllByLabelText} = render(<Component/>)
+        queryAllByLabelText(/hello/i)`,
+    },
+    {
+      code: `
+        import { within } from '@testing-library/dom'
+        within(element).findByRole('button', {name: /hello/ig})`,
```

Comment:
Could you swap the `ig` regexp as to `gi` in one of these tests, so we also make sure the order of the params don't affect the autofix?

Reply:
`gi` and `ig` are already used in the invalid cases, I'll convert this to a regexp that uses 3 flags.

Reply:
Ooh my bad! I didn't realize about that. Good idea about including the other format.